packages/app-core/src/modules/commentsArea/usecases/fetchCommentsAreas/fetchCommentsAreas.ts

```diff
@@ -1,17 +1,23 @@
-import { commentsAreaDtoToEntity } from '../../../../entities';
+import { CommentsAreaDto, commentsAreaDtoToEntity } from '../../../../entities';
+import { Paginated } from '../../../../shared';
 import { createThunk } from '../../../../store/createThunk';
 import { setCommentsArea } from '../../../../store/normalize';
 import { setCommentsAreas, setFetchingCommentsAreas, setTotalCommentsAreas } from '../../actions';
-import { selectTotalCommentsAreas } from '../../selectors';
-
-export const fetchCommentsAreas = createThunk(async ({ getState, dispatch, commentsAreaGateway }) => {
-  if (selectTotalCommentsAreas(getState()) > 0) {
-    return;
-  }
+import { selectCommentsAreasSearchQuery } from '../../selectors';
 
+export const fetchCommentsAreas = createThunk(async ({ getState, dispatch, commentsAreaGateway }, query?: string) => {
   dispatch(setFetchingCommentsAreas(true));
 
-  const commentsAreasDtos = await commentsAreaGateway.fetchCommentsAreas();
+  const search = query ?? selectCommentsAreasSearchQuery(getState());
```

Comment:
What about removing this use case's `query` parameter, and always use the value from the store?

Reply:
I think it's a good idea. No need of source duplication :)